### PR TITLE
Prettier xml

### LINKED ISSUE #103: Prettier XML

### PR DESCRIPTION
**Pull request by MSaguer**
To close #103 

Prettier-xml added and called automatically during commit with default values

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
I don't understand how this plugin should work.
I changed the indentation in `pom.xml` :

image

But everything is well formated:

image

**Comment by MSaguer** [comment]
That also surprised me that this indentation feature doesn't work out of
the box.
I must have missed something. I'll give it another try in the week.

Le mar. 21 déc. 2021 à 22:15, Pascal Grimaud ***@***.***> a
écrit :

**Comment by pascalgrimaud** [comment]
@MSaguer : I found the issue. We need to add this property in `.prettierrc` file

```diff
diff --git a/.prettierrc b/.prettierrc
index 34b8d15..7c210c3 100644
--- a/.prettierrc
+++ b/.prettierrc
@@ -11,6 +11,9 @@ arrowParens: avoid
 # jsx and tsx rules:
 bracketSameLine: false
 
+# xml
+xmlWhitespaceSensitivity: ignore
+
 # java rules:
 overrides:
   - files: "*.java"
```

File: package.json [hunk 1]
```diff
@@ -6,13 +6,14 @@
   "license": "UNLICENSED",
   "scripts": {
     "prepare": "husky install",
-    "prettier:check": "prettier --check \"{,src/**/}*.{md,json,yml,html,js,ts,tsx,css,scss,vue,java}\"",
-    "prettier:format": "prettier --write \"{,src/**/}*.{md,json,yml,html,js,ts,tsx,css,scss,vue,java}\""
+    "prettier:check": "prettier --check \"{,src/**/}*.{md,json,yml,html,js,ts,tsx,css,scss,vue,java,xml}\"",
+    "prettier:format": "prettier --write \"{,src/**/}*.{md,json,yml,html,js,ts,tsx,css,scss,vue,java,xml}\""
   },
   "devDependencies": {
+    "@prettier/plugin-xml": "^1.1.0",
```
Comment (pascalgrimaud) [inline package.json:13]: I'd prefer to use fixed version, as we have tons of issue in generator-jhipster when using `^`

```suggestion
    "@prettier/plugin-xml": "1.1.0",
```

File: package.json [hunk 2]
```diff
@@ -6,13 +6,14 @@
...
     "husky": "7.0.4",
     "lint-staged": "12.1.3",
-    "prettier": "2.5.1",
+    "prettier": "^2.5.1",
```
Comment (pascalgrimaud) [inline package.json:16]:
```suggestion
    "prettier": "2.5.1",
```

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

**Review by pascalgrimaud — APPROVED** [review]

**Pull request merged**